lib/phoenix/endpoint/adapter.ex

```diff
@@ -4,26 +4,45 @@ defmodule Phoenix.Endpoint.Adapter do
   # the adapters/handlers.
   @moduledoc false
 
+  alias Phoenix.CodeReloader.ChangeDetector
+
   @doc """
   Starts the endpoint supervision tree.
   """
   def start_link(otp_app, mod) do
     import Supervisor.Spec
-    pub_conf = config(otp_app, mod)[:pubsub]
+    conf       = config(otp_app, mod)
+    pub_conf   = conf[:pubsub]
 
     pubsub_children = case pub_conf[:adapter] do
       nil     -> []
       adapter -> [supervisor(adapter, [mod.__pubsub_server__(), pub_conf])]
     end
 
-    children = pubsub_children ++ [
+    watcher_children =
+      Enum.reduce(conf[:watchers], [], fn {cmd, args}, acc ->
```

Comment:
Yes. Initially I was joining the live reload + watchers as one list, so this can just be a map now